src/sparseml/pytorch/torchvision/train.py

```diff
@@ -473,13 +506,20 @@ def collate_fn(batch):
                 "model": model_without_ddp.state_dict(),
                 "optimizer": optimizer.state_dict(),
                 "lr_scheduler": lr_scheduler.state_dict(),
-                "epoch": epoch,
+                "epoch": -1 if epoch == manager.max_epochs - 1 else epoch,
```

Comment:
epoch here should factor in composed staged (ie add the max epoch for the checkpoint recipe)